### DISCUSSION TITLE
Start using GitHub Projects to plan tedana releases and/or track progress

### DISCUSSION THREAD

**Discussion by eurunuela** [ME-ICA/tedana#988] [Ideas]
I was wondering if you think using [GitHub Projects](https://github.blog/2022-07-27-planning-next-to-your-code-github-projects-is-now-generally-available/) would help us organize our work and efforts better.

I know we are not very consistent with our contributions, but I think this is also in part due to our lack of organization when it comes to prioritizing what to put our efforts into and when to do it.

I believe that using a board would help us be more productive and deliver more consistently.

When it comes to how we would use it, I see it working in two possible ways:
- Every action item from our meetings becomes a GitHub issue, and we have a monthly board to see what we want to get done for the coming meeting. This would allow us to better see progress every month (and it would probably make it easier to populate our meeting agenda).
- We make quarterly releases (or whatever) and have a board for every release, with clear goals (issues) we want to tackle before merging. This would make it very clear when we are ready to cut a release and the progress made every quarter (or whatever).

I would like to know what others think, especially those who have been most actively writing code in tedana (@handwerkerd and @tsalo).

Edit: I would be happy to beta test this (as in, set it up and make sure it's up to date), but I do not have permission to open a public project within ME-ICA.

**Comment by handwerkerd** [comment]
I do try to turn everything we discuss at a meeting into an issue so that part is already done. We've used projects in the past, but the trouble is it's one more thing to keep up to date. (For example see our current two open projects, which should both be closed  https://github.com/ME-ICA/tedana/projects?type=classic ). I'm game for trying again. If it helps us keep track of things and proactively prioritize, that would be wonderful, but my worry is that it will be extra work to retroactively update something we're not as actively tracking.

Projects might also help clean up the agenda because I wouldn't need to use the agenda to list some of the standing issues that we aren't actively working on, but we don't want to let completely slip.

As for what the projects could be, I could see one on Documentation/Education which would never close. That would probably be best under meica rather than meica/tedana. As you wrote, another one for a version release makes sense.

**Reply by eurunuela** [reply]
I can create them if you want. I will need some GitHub permission for that though. At the moment I can only create private projects.

**Reply by eurunuela** [reply]
I also wanted to mention that we can [automate the project boards quite a bit using GitHub Actions](https://github.com/actions/add-to-project).

**Reply by handwerkerd** [reply]
I think I just gave permission for members to make private projects public. There's an option to give all members "admin" access for all projects, but I'm not sure that's necessary. If my current change didn't work, I'm fairly sure, if you create a private project and invite me, I'll be able to make it public.
